### PR TITLE
newdevice: Add support for KTI switches

### PR DESCRIPTION
**Pull request by corny**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you signed the [Contributors agreement](http://docs.librenms.org/General/Contributing/)
- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

### CODE REVIEW

File: tests/snmpsim/kti.snmprec [hunk 1]
```diff
@@ -0,0 +1 @@
+1.3.6.1.2.1.1.1.0|4|KGS-1060 GigaBit Ethernet Switch
```
Comment (murrant) [inline tests/snmpsim/kti.snmprec]: We need both sysDescr and sysObjectId

File: includes/polling/os/kti.inc.php [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?php
+
+$hardware     = trim(snmp_get($device, ".1.3.6.1.2.1.1.1.0", '-Ovq'), '"');
```
Comment (murrant) [inline includes/polling/os/kti.inc.php]: sysDescr is already fetched, you can use `$poll_device['sysDescr']`

File: tests/snmpsim/kti.snmprec (hunk 1 above)
Comment (laf) [inline tests/snmpsim/kti.snmprec]: It's also worth switching the yaml discover to use sysObjectId if it's useful. Relying on sysDescr matches should be the last resort.

Comment (corny) [inline tests/snmpsim/kti.snmprec]: The sysObjectId is not useful:
```
$ snmpbulkwalk -OntQX -v2c -c public $host 1.3.6.1.2.1.1
.1.3.6.1.2.1.1.1.0 = "KGS-1060 GigaBit Ethernet Switch"
.1.3.6.1.2.1.1.2.0 = .0.0
```

Comment (laf) [inline tests/snmpsim/kti.snmprec]: Shoot that vendor

File: includes/polling/os/kti.inc.php [hunk 3]
```diff
@@ -0,0 +1,4 @@
+<?php
+
+$hardware     = trim(snmp_get($poll_device['sysDescr'], ".1.3.6.1.2.1.1.1.0", '-Ovq'), '"');
```
Comment (laf) [inline includes/polling/os/kti.inc.php]: What murrant meant here was that $poll_device['sysDescr'] already contains the output you want so you don't need another snmp call just $hardware = $poll_device['sysDescr']

**Comment by laf** [comment]
I've pushed the update.

**Pull request merged**